src/datepicker/utils/index.js

```diff
@@ -6,256 +6,52 @@ LICENSE file in the root directory of this source tree.
-    excludeDates: ?Array<Date>,
-    includeDates: ?Array<Date>,
-    filterDate: ?(day: Date) => boolean,
-  } = {},
-) {
-  return (
-    isOutOfBounds(day, {minDate, maxDate}) ||
-    (excludeDates &&
-      excludeDates.some(excludeDate => isSameDay(day, excludeDate))) ||
-    (includeDates &&
-      !includeDates.some(includeDate => isSameDay(day, includeDate))) ||
-    (filterDate && !filterDate(day)) ||
-    false
-  );
-}
-
-export function isOutOfBounds(
-  day: Date,
-  {minDate, maxDate}: {minDate: ?Date, maxDate: ?Date} = {},
-) {
-  return (
-    (minDate && differenceInCalendarDays(day, minDate) < 0) ||
-    (maxDate && differenceInCalendarDays(day, maxDate) > 0)
-  );
-}
-
-export function monthDisabledBefore(
-  day: Date,
-  {minDate, includeDates}: {minDate: ?Date, includeDates: ?Array<Date>} = {},
-) {
-  const previousMonth = subMonths(day, 1);
-  return (
-    (minDate && differenceInCalendarMonths(minDate, previousMonth) > 0) ||
-    (includeDates &&
-      includeDates.every(
-        includeDate =>
-          differenceInCalendarMonths(includeDate, previousMonth) > 0,
-      )) ||
-    false
-  );
-}
-
-export function monthDisabledAfter(
-  day: Date,
-  {maxDate, includeDates}: CalendarPropsT = {},
-) {
-  const nextMonth = addMonths(day, 1);
-  return (
-    (maxDate && differenceInCalendarMonths(nextMonth, maxDate) > 0) ||
-    (includeDates &&
-      includeDates.every(
-        includeDate => differenceInCalendarMonths(nextMonth, includeDate) > 0,
-      )) ||
-    false
-  );
-}
-
-export function getEffectiveMinDate({minDate, includeDates}: CalendarPropsT) {
-  if (includeDates && minDate) {
-    let minDates = includeDates.filter(
-      includeDate => differenceInCalendarDays(includeDate, minDate) >= 0,
-    );
-    // $FlowFixMe
-    return min(minDates);
-  } else if (includeDates && includeDates.length) {
-    // $FlowFixMe
-    return min(includeDates);
-  } else {
-    return minDate;
-  }
-}
-
-export function getEffectiveMaxDate({maxDate, includeDates}: CalendarPropsT) {
-  if (includeDates && maxDate) {
-    let maxDates = includeDates.filter(
-      includeDate => differenceInCalendarDays(includeDate, maxDate) <= 0,
-    );
-    // $FlowFixMe
-    return max(maxDates);
-  } else if (includeDates) {
-    // $FlowFixMe
-    return max(includeDates);
-  } else {
-    return maxDate;
-  }
-}
-
-export function applyTimeToDate(date: ?Date, time: Date) {
-  if (!date) return time;
-  return set(date, {
-    hours: getHours(time),
-    minutes: getMinutes(time),
-    seconds: 0,
-  });
-}
-
-export function applyDateToTime(time: ?Date, date: Date) {
-  if (!time) return date;
-  return set(time, {
-    year: getYear(date),
-    month: getMonth(date),
-    date: getDate(date),
-  });
-}
+import defaultAdapter from './date-fns-adapter';
+import DateHelpers from './date-helpers';
+
+const defaultDateHelpers = new DateHelpers<Date>(defaultAdapter);
+
+export const formatDate = defaultDateHelpers.formatDate;
```

Comment:
I couldn't think of a way to have all of these be exported as named exports without explicitly defining them like this